### PR TITLE
Implemented trie datastructure in Java

### PR DESCRIPTION
**Pull request by samar08**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes

### Description

This file contains code for Trie datastructure implemented in Java. 

### Checklist

- [√ ] I've read the contribution guidelines.
- [√ ] I've checked the issue list before deciding what to submit.
- [√ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Issue number: #2301

### CODE REVIEW

**Comment by samar08** [comment]
Please merge my pull request. Thanks and regards: A.M.Samara simha reddy

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Remove the comment you have added at the top
- Add a beginner-friendly description at the top
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

**Comment by samar08** [comment]
> Few changes:
> 
> * Remove the comment you have added at the top
> * Add a beginner-friendly description at the top
> * Remove extra lines of code and spaces
> * Add sample I/O at the end of the Code
> * Add Time and Space Complexity at the end of the Code

I have made the required changes, please verify them.

File: Java/ds/Trie.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie.java]: Remove the line or space.

Comment (ankitaggarwal23) [inline Java/ds/Trie.java]: Remove the line or space.

File: Java/ds/Trie.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    void addelement(String element) {
+        TrieNode temp = root;
+        int i, n = element.length();
+        Character s;
+        for (i = 0; i < n; i++) {
+            s = element.charAt(i);
+            if (temp.map.containsKey(s)) {
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie.java]: Remove the line or space.

File: Java/ds/Trie.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                temp = temp.map.get(s);
+            } else {
+                temp.map.put(s, new TrieNode());
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie.java]: Remove the line or space.

File: Java/ds/Trie.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+}
+/*
+ * Sample Input: Please enter the size of the input Array N: 5 Please enter the
+ * words in next N lines samara simha reddy attunuru narasimha Please enter
+ * number of words you want to search in trie M: 5 Enter the word to search in
+ * trie: samara Found Enter the word to search in trie: samar Not Found Enter
+ * the word to search in trie: simhas Not Found Enter the word to search in
+ * trie: simha Found Enter the word to search in trie: nara Not Found
```
Comment (ankitaggarwal23) [inline Java/ds/Trie.java]: Your I/O is not readable. Use newline and space after reading input or print any output.

File: Java/ds/Trie.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ * 
+ * Time complexity: O(M) for searching a word of length M.
+ *
```
Comment (ankitaggarwal23) [inline Java/ds/Trie.java]: Remove the line or space.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by ankitaggarwal23** [comment]
@samar08 As I'm seeing you closed this issue [here](https://github.com/TesseractCoding/NeoAlgo/issues/2301) and after that we assigned it to someone else [here](https://github.com/TesseractCoding/NeoAlgo/issues/2968). So we can't review this PR again as it is already assigned to someone. Thanks, Closing this is issue.😊

**Pull request closed without merging**